src/openprocurement/api/utils.py

```diff
@@ -793,29 +821,30 @@ def calculate_business_date(date_obj, timedelta_obj, context=None, working_days=
     :type working_days: bool
     :return: the end of period
     :rtype: datetime.datetime
+
     """
 
-    # Acceleration mode logic
-    if context and 'procurementMethodDetails' in context and context['procurementMethodDetails']:
-        re_obj = ACCELERATOR_RE.search(context['procurementMethodDetails'])
-        if re_obj and 'accelerator' in re_obj.groupdict():
-            return date_obj + (timedelta_obj / int(re_obj.groupdict()['accelerator']))
-
-    if working_days:
-        if timedelta_obj > timedelta():
-            if is_holiday(date_obj):
-                date_obj = datetime.combine(date_obj.date(), time(0, tzinfo=date_obj.tzinfo)) + timedelta(1)
-                while is_holiday(date_obj):
-                    date_obj += timedelta(1)
+    accelerated_calculation = accelerated_calculate_business_date(date_obj, timedelta_obj, context)
+    if accelerated_calculation:
+        return accelerated_calculation
+
+    if not working_days:
+        return date_obj + timedelta_obj
+
+    # reset datetime to exclude influence of time data (e.g. hour, minute) on calculations
+    date_obj = reset_to_start_of_the_day(date_obj)
```

Comment:
Не до кінця розумію. Нам потрібно скинути час, щоб він не спричинив помилки в розрахунках. Вжитий метод виконує цю роботу + має по собі документацію і явно зберігає часовий пояс даного `date_obj`.

Reply:
Чи можемо ми робити розрахунки з date об'єктом, а не datetime
Тобто добавляти дні до `yyyy-mm-dd`, а не до `yyyy-mm-ddThh:mm+tttt` ?

Reply:
З одного боку можемо, напевне, але на об'єкті, який повертає ця функція, залежить дуже багато коду, де до нього відносяться як до `datetime`, а не як до `date`, зокрема `chronograph`. Тому оце "напевне" може спричинити важковідловлювані проблеми.
Думаю, варто спробувати попрацювати з цією відрефактореною версію коду: чи часом він не спричинить якихось проблем, а на 847 рядку поставити TODO типу такого:
`# TODO (after 1-8-2018) switch to date object instead of datetime object`

Reply:
Тоді доповни #TODO і будемо приймати